src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,119 @@
+                tasks.add(newTask);
+                System.out.println("  " + newTask.toString());
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else if (userInput.split(" ")[0].equals("deadline")) {
+                if(userInput.length() < 10) {
+                    Output.printEmptyCommand("deadline");
+                    continue;
+                }
+                System.out.println("Got it. I've added this task: ");
+                String by = "";
+                String processedDeadlineInput;
+                int getSlashIndex = 0;
+                for (int i = 0; i < userInput.length(); i++) {
+                    char getSlash = userInput.charAt(i);
+                    if (getSlash == '/') {
+                        getSlashIndex = i;
+                        break;
+                    }
+                }
+                by = userInput.substring(getSlashIndex + 4);
+                processedDeadlineInput = userInput.substring(9, getSlashIndex).trim();
+                Task newTask = new Deadline(processedDeadlineInput, by);
+                tasks.add(newTask);
+                System.out.println("  " + newTask.toString());
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else if (userInput.split(" ")[0].equals("event")) {
+                if(userInput.length() < 7) {
+                    Output.printEmptyCommand("event");
+                    continue;
+                }
+                System.out.println("Got it. I've added this task: ");
+                String at;
+                String processedEventInput;
+                int getSlashIndex = 0;
+                for (int i = 0; i < userInput.length(); i++) {
+                    char getSlash = userInput.charAt(i);
+                    if (getSlash == '/') {
+                        getSlashIndex = i;
+                        break;
+                    }
+                }
+                at = userInput.substring(getSlashIndex + 4).trim();
+                processedEventInput = userInput.substring(6, getSlashIndex).trim();
+                Task newTask = new Event(processedEventInput, at);
+                tasks.add(newTask);
+                System.out.println("  " + newTask.toString());
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else if (userInput.split(" ")[0].equals("delete")) {
+                if(userInput.length() < 8) {
+                    Output.printEmptyCommand("delete");
+                    continue;
+                } else {
+                    int processedInput;
+                    processedInput = Integer.parseInt(userInput.replaceAll("[^0-9]", "")) - 1;
+                    if(processedInput < 0 || processedInput > tasks.size()) {
+                        System.out.println("No index found. Please key an appropriate index");
```

Comment:
Perhaps consider using SLAP and refactor this into a method. This will reduce nesting in the code and make it more readable.